bat.d/66-wilco-ec

```diff
@@ -0,0 +1,730 @@
+    #       0=thresholds/
+    #       32=disabled/
+    #       128=no kernel support/
+    #       254=laptop not supported
+    #
+    # 2. determine method for
+    #    reading battery data                   --> retval $_bm_read,
+    #    reading/writing charging thresholds    --> retval $_bm_thresh,
+    #    reading/writing force discharge        --> retval $_bm_dischg:
+    #       none/natacpi
+    #
+    # 3. define sysfile basenames for natacpi
+    #    start threshold                        --> retval $_bn_start,
+    #    stop threshold                         --> retval $_bn_stop,
+    #    discharge                              --> retval $_bn_discharge
+    #    charge type (Dell specific)            --> retval $_bn_chtype
+    #
+    # 4. determine present batteries
+    #    list of batteries (space separated)    --> retval $_batteries;
+    #
+    # 5. define charge threshold defaults
+    #    start threshold                        --> retval $_bt_def_start,
+    #    stop threshold                         --> retval $_bt_def_stop;
+
+    _batdrv_plugin="wilco-ec"
+    _batdrv_kmod="wilco_charger" # kernel module for natacpi
+    _batdrv_sim=0
+
+    # check plugin simulation override and denylist
+    if [ -n "$X_BAT_PLUGIN_SIMULATE" ]; then
+        if [ "$X_BAT_PLUGIN_SIMULATE" = "$_batdrv_plugin" ]; then
+            echo_debug "bat" "batdrv_init.${_batdrv_plugin}.simulate"
+            _batdrv_sim=1
+        else
+            echo_debug "bat" "batdrv_init.${_batdrv_plugin}.simulate_skip"
+            return 1
+        fi
+    elif wordinlist "$_batdrv_plugin" "$X_BAT_PLUGIN_DENYLIST"; then
+        echo_debug "bat" "batdrv_init.${_batdrv_plugin}.denylist"
+        return 1
+    else
+        # check if hardware matches
+        if ! batdrv_is_dell; then
+            echo_debug "bat" "batdrv_init.${_batdrv_plugin}.no_match"
+            return 1
+        fi
+    fi
+
+    # presume no features at all
+    _natacpi=128
+    _bm_read="natacpi"
+    _bm_thresh="none"
+    _bm_dischg="none"
+    _bn_start=""
+    _bn_stop=""
+    _bn_dischg=""
+    _bn_chtype=""
+    _batteries=""
+    _bt_def_start=95
+    _bt_def_stop=100
+
+    # iterate batteries and check for native kernel ACPI
+    local bd bs
+    local done=0
+    for bd in "$ACPIBATDIR"/BAT[01]; do
+        if [ "$(read_sysf "$bd/present")" = "1" ]; then
+            # record detected batteries and directories
+            bs=${bd##/*/}
+            if [ -n "$_batteries" ]; then
+                _batteries="$_batteries $bs"
+            else
+                _batteries="$bs"
+            fi
+            # skip natacpi detection for 2nd and subsequent batteries
+            [ $done -eq 1 ] && continue
+
+            done=1
+            if [ "$NATACPI_ENABLE" = "0" ]; then
+                # natacpi disabled in configuration --> skip actual detection
+                _natacpi=32
+                continue
+            fi
+
+            bd="$ACPIBATDIR/wilco-charger"
```

Comment:
Move the following block out of the bd loop . Change the _bn's to _bf's and assign the absolute path.